src/planscape/forsys/views.py

```diff
@@ -3,43 +3,321 @@
+
+  import rpy2.robjects as robjects
+  robjects.r.source(os.path.join(settings.BASE_DIR, 'forsys/scenario_sets.R'))
+  scenario_sets_function_r = robjects.globalenv['scenario_sets']
+ 
+  # TODO: add inputs for thresholds.
+  # TODO: clean-up: pass header names (e.g. proj_id) into scenario_sets_function_r.
+  rdf = convert_dictionary_of_lists_to_rdf(npdf)
+
+  priority_headers = []
+  for p in priorities:
+    priority_headers.append(kPriorityPrefix + "_" + p)
+
+  forsys_output = scenario_sets_function_r(rdf,
+                                           robjects.FloatVector([np.ceil(np.sum(npdf['area']))]),
+                                           robjects.StrVector(priority_headers))
+
+  # TODO: add logic for applying constraints to forsys_output.
+
+  result = {}
+  result['top_stands'] = convert_rdf_to_pddf(forsys_output[0])
+  result['top_projects'] = convert_rdf_to_pddf(forsys_output[1])
+  return result
+
 # Returns JSon data for a forsys scenario set call.
 def scenario_set(request: HttpRequest) -> HttpResponse:
   try:
-    target_value = get_target_value(request.GET)
-    import rpy2.robjects as robjects
-    robjects.r.source(os.path.join(settings.BASE_DIR, 'forsys/dummy.R'))
-    dummy_function_r = robjects.globalenv['dummy']
+    # TODO: fetch region, priorities, stand type, and project area as url parameters.
+    save_debug_info = True
+    region = 'sierra_cascade_inyo'
+    priorities = ['fire_dynamics', 'forest_resilience', 'species_diversity']
+
+    huc12_id = 43
+
+    project_area = Polygon( ((-120.14015536869722, 37.05413814388948),
+                             (-120.18409937110482, 36.9321584213366),
+                             (-119.93422142411087, 36.94003252840713),
+                             (-120.03710286062301, 36.99713288358574),
+                             (-120.14015536869722, 37.05413814388948)) )
+    project_area.srid = 4269
+    if not project_area.valid:
+      raise ValueError("invalid project area: %s"%project_area.valid_reason) 
 
-    analysis = dummy_function_r(target_value)
+    # raster boundary seems to be 80 units off from the huc-12 boundaries.
+    # TODO: adjust raster data.
+    project_area_translated = translate_polygon(project_area, 80, 0)
+    project_area_raster = project_area_translated.clone()
+    project_area_raster.transform(CoordTransform(SpatialReference(4269), SpatialReference(9822)))
 
-    # TODO: clarify output variables.
     response = {}
-    for i in range(len(analysis)):
-      pddata = convert_rdf_to_pddf(analysis[i])
-      response['type_%d'%(i)] = pddata.to_json()
-    return JsonResponse(response)
+    if (save_debug_info):
+      response['debug'] = {}
+
+    # Filters boundaries by boundary_id.
+    # TODO: add more stand options. For the existing solution, project areas drawn manually are divided into stands according to HUC-12 boundaries.
+    boundaries = BoundaryDetails.objects.filter(boundary_id=huc12_id).filter(geometry__intersects=project_area)
```

Comment:
if we use intersect here we will have stands that end up in multiple projects. IMO we should use centroid instead. That way each cell (stand) can belong to exactly one project only (for non-overlapping projects)

Reply:
Yes: I think we can have multiple stand options, including centroid, raster pixel, etc. Let's check this in so that we have a solution that can be further refined by everyone.

Downstream, a polygon representing the intersection between these HUC-12 boundaries and a project area is computed as the "stand". So we will not have stands that wind up in multiple project areas.